### PR TITLE
add Promin@v1.0.1 w/ git auto-update

### LINKED ISSUE #8525: [Request] Add Promin
**Library name:** Promin
**Git repository url:** https://github.com/timseverien/Promin

### PR DESCRIPTION
**Pull request by nolsherry**
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/timseverien/Promin
 * Official website (optional, not the repository):
 * NPM package url (optional):
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 208
   - Count of watchers: 15
   - Count of forks: 33
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [X] This lib was not found on cdnjs repo
 * [X] No already exist / duplicated issue and PR
 * [X] The lib has notable popularity
   * [X] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [X] Has valid tags for each versions (for git auto-update)
 * [X] Auto-update setup
 * [X] Auto-update target/source is valid.
 * [X] Auto-update filemap is correct.

# Git commit checklist
 * [X] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [X] The parent of the commit(s) in the PR is not old than 3 days.
 * [X] Pull request is sending from a non-master branch with meaningful name.
 * [X] Separate unrelated changes into different commits.
 * [X] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [X] Close corresponding issue in commit message
 * [X] Mention related issue(s), people in commit message, comment.

close #8525, cc @timseverien

### CODE REVIEW

**Comment by nolsherry** [comment]
@PeterDaveHello, I have used [web-minify-helper/minify.sh](https://github.com/PeterDaveHello/web-minify-helper) to minify `js/jquery.promin.js` and `css/promin.css`, but it seem to be useless.
Could you help me fix this problem? thanks!

**Comment by PeterDaveHello** [comment]
@nolsherry You mean css file still been exluded?

**Comment by nolsherry** [comment]
@PeterDaveHello, I mean after using [web-minify-helper/minify.sh](https://github.com/PeterDaveHello/web-minify-helper) to minify 
`js/jquery.promin.js` and `css/promin.css`, it can not generate `js/jquery.promin.min.js` and `css/promin.min.css`, thanks!

**Comment by PeterDaveHello** [comment]
@nolsherry I thought you've found a bug, and it was fixed in https://github.com/PeterDaveHello/web-minify-helper/commit/f287dacdbc7b5674c7cb4e49cc6673bcedb404b4, thanks.

**Comment by nolsherry** [comment]
@PeterDaveHello, thanks for your help. I have minified these files successfully.
@kennynaoh, @pvnr0082t, please help me review this PR, thanks!

File: ajax/libs/Promin/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "plugin",
+    "html"
+  ],
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/timseverien/Promin.git",
+    "basePath": "",
+    "files": [
+      "css/*",
+      "js/*"
```
Comment (pvnr0082t) [inline ajax/libs/Promin/package.json]: We can use `+(css|js)/*promin*` here

**Comment by pvnr0082t** [comment]
@nolsherry Any reason you minified files without generating `.map` file?

**Comment by nolsherry** [comment]
@pvnr0082t , @kennynaoh , I have modified the filemap of `package.json` and generate `.map` file.
Please help me review this PR, thanks!

**Comment by PeterDaveHello** [comment]
@nolsherry I think you should "answer" the question before do something else like fixing, that's how "discussing" works, otherwise no body knows what happened here.

**Comment by nolsherry** [comment]
@pvnr0082t, after I reported the problem of `minify.sh` to @PeterDaveHello, I pull the latest version of `minify.sh` and it can minify ccs/jquery.promin.js and css/promin.css. However, I seems that there does not generate `.map` file, therefore,  I pull the latest version of [web-minify-helper](https://github.com/PeterDaveHello/web-minify-helper) to minify these files again.

**Comment by pvnr0082t** [comment]
LGTM

**Comment by PeterDaveHello** [comment]
@nolsherry I don't think any of the updates have something to do with map file.

**Comment by PeterDaveHello** [comment]
LGTM

**Pull request merged**